File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Shortcut.java
package com.faendir.lightning_launcher.multitool.proxy;

import android.content.Intent;

/**
 * @author lukas
 * @since 04.07.18
 */
public interface Shortcut extends Item {
    String getLabel();

    void setLabel(String label);

    Intent getIntent();

    Image getDefaultIcon();

    Image getCustomIcon();

    void setDefaultIcon(Image image);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Shortcut.kt
package com.faendir.lightning_launcher.multitool.proxy

import android.content.Intent

/**
 * @author lukas
 * @since 04.07.18
 */
interface Shortcut : Item {
    var label: String

    val intent: Intent

    var defaultIcon: Image

    val customIcon: Image
}
